nengo/utils/connection.py

```diff
@@ -107,3 +107,21 @@ def eval_point_decoding(conn, sim, eval_points=None):
     decoded = np.dot(activities, weights.T)
     targets = get_targets(conn, eval_points)
     return eval_points, targets, decoded
+
+
+def function_name(func):
+    """Returns the name of a connection function (robust to the different
+    types of objects that can be passed as a function argument).
+
+    Parameters
+    ----------
+    func : callable or array-like
+        object used as function argument
+
+    Returns
+    -------
+    str
+        name of function object
+    """
+
+    return getattr(func, "__name__", func.__class__.__name__)
```

Comment:
Looking at the commit though, I did it to help with some old types of objects. I wish I could remember what, exactly. https://github.com/nengo/nengo/pull/1010/commits/76f0a34571796e07fcd3be9db62f68e12cf007da